### PR TITLE
🚀  Adding Create Post Form

### LINKED ISSUE #92: 🚀  Adding Create Post Form
In order for users to publish posts onto the feed page we need to add a form that allows them to submit posts.

Steps to follow:
- Add a CreateView
- Add a HTML template
- Set a page url
- Add relevant tests

### PR DESCRIPTION
**Pull request by paOmer**
Added a post creation form to let users have the ability to publish posts.
Used django [CreateView](https://docs.djangoproject.com/en/4.0/ref/class-based-views/generic-editing/#createview) class.

- Added a CreateView for a new post.
- Added a redirection function on the post model to be used after creating new post.
- Set the post creation page url.
- Added a post creation html template.
- Added relevant tests.

I tried to not include the post creation html template in this PR in order to not include UI but I had to set a html page for testing purposes.

Future tasks:
- Adding the ability to create a job-offer post (#96)
- Adding a link to the create new post form (#95)
- Style the form using crispy-forms (#94)

Close #92

### CODE REVIEW

**Comment by Yarboa** [comment]
> Added a post creation form to let users have the ability to publish posts. Used django [CreateView](https://docs.djangoproject.com/en/4.0/ref/class-based-views/generic-editing/#createview) class.
> 
> * Added a CreateView for a new post.
> * Added a redirection function on the post model to be used after creating new post.
> * Set the post creation page url.
> * Added a post creation html template.
> * Added relevant tests.
> 
> I tried to not include the post creation html template in this PR in order to not include UI but I had to set a html page for testing purposes.
> 
> Future tasks:

Please add an Issue here
> 
> * Adding the ability to create a job-offer post
> * Adding a link to the create new post form
> * Style the form using crispy-forms
> 
> Close #92

File: feed/views.py [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 
 
 def feed(request):
     context = {
         'posts': Post.posts.main_feed()
     }
     return render(request, 'feed/feed.html', context)
+
+
+class PostCreateView(LoginRequiredMixin, CreateView):
```
Comment (Yarboa) [inline feed/views.py:14]: Suggest to follow documentation
https://docs.djangoproject.com/en/4.0/topics/auth/default/#how-to-log-a-user-in-1

**Review by Yarboa — CHANGES_REQUESTED** [review]

File: feed/views.py (hunk 1 above)
Comment (paOmer) [reply feed/views.py:14]: Im not sure what to change here in `PostCreateView` but I've applied the suggestions to the `test_authenticated_users_access` test.
Would be happy for more guidance

**Comment by paOmer** [comment]
> > Added a post creation form to let users have the ability to publish posts. Used django [CreateView](https://docs.djangoproject.com/en/4.0/ref/class-based-views/generic-editing/#createview) class.
> > 
> > * Added a CreateView for a new post.
> > * Added a redirection function on the post model to be used after creating new post.
> > * Set the post creation page url.
> > * Added a post creation html template.
> > * Added relevant tests.
> > 
> > I tried to not include the post creation html template in this PR in order to not include UI but I had to set a html page for testing purposes.
> > Future tasks:
> 
> Please add an Issue here
> 
> > * Adding the ability to create a job-offer post
> > * Adding a link to the create new post form
> > * Style the form using crispy-forms
> > 
> > Close #92

Created the issues and edited this PR description to include them.

File: feed/models/post.py [hunk 2]
```diff
@@ -17,3 +18,6 @@ class Post(models.Model):
 
     def __str__(self):
         return self.title
+
+    def get_absolute_url(self):
+        return reverse('feed')
```
Comment (ShirleyFichman) [inline feed/models/post.py:23]: @paOmer could you please explain why is this needed, do we need to add this in every app?

Comment (paOmer) [reply feed/models/post.py:23]: In this case the create new post form has been created with Django's CreateView class.
After successfully submitting such a form, the CreateView class is looking to redirect the user to another page in the website, its default way is to look for such a function in the model file.

This do not need to be added to every app or model in the project, only the ones that are relevant.

Comment (ShirleyFichman) [reply feed/models/post.py:23]: I understand, thank you for the explanation.

**Review by ShirleyFichman — APPROVED** [review]
LGTM

**Review by DeanBiton — APPROVED** [review]
LGTM

File: feed/views.py (hunk 1 above)
Comment (Yarboa) [reply feed/views.py:19]: Maybe this will give you one tests idea
https://github.com/redhat-beyond/beyond-tutorial/blob/main/msgboard/tests.py#L25-L29

And this 
https://github.com/redhat-beyond/beyond-tutorial/blob/main/msgboard/tests.py#L47-L51

Look at the whole login under views
https://github.com/redhat-beyond/beyond-tutorial/blob/main/msgboard/views.py#L7

Suggested to work as per the docs, Stupid Simple

File: feed/views.py [hunk 3]
```diff
@@ -13,3 +14,13 @@ def feed(request):
...
 class PostDetailView(DetailView):
     model = Post
+
+
+class PostCreateView(LoginRequiredMixin, CreateView):
+    model = Post
+    fields = ['title', 'content']
+    login_url = '/login/'
+
+    def form_valid(self, form):
```
Comment (Yarboa) [inline feed/views.py:24]: Look at the samples above, there is recommended way to do it

Comment (paOmer) [reply feed/views.py:24]: Done

File: feed/views.py (hunk 1 above)
Comment (Yarboa) [reply feed/views.py:35]: Yes, but NVM. there is different way to do that

**Review by Yarboa — COMMENTED** [review]
@paOmer Nice work, I just do not see any client tests in the PR.
When we test View/Form it is not guaranteed the the hole path is not broken
from http request to http response

**Comment by paOmer** [comment]
> @paOmer Nice work, I just do not see any client tests in the PR. When we test View/Form it is not guaranteed the the hole path is not broken from http request to http response

Updated the tests accordingly.

**Review by Yarboa — APPROVED** [review]
LGTM
